### PR TITLE
add docs for endpoints GET /ad/count and /ad/histogram

### PR DESCRIPTION
**Pull request by twessling-icas**
idk how to properly test this locally except for using the intellij openapi preview pane... any hints to test in real browser from local?

### CODE REVIEW

File: openapi/openapi.yaml [hunk 1]
```diff
@@ -1233,6 +1233,303 @@ paths:
...
+              format: int64
+          example: [ 11232, 2231, 32453 ]
+        - name: titleKeywords
+          in: query
+          required: false
+          description: 'Case-insensitive filter for a keyword in the title; will include only ads which contain the given keyword in their title.'
+          schema:
+            type: string
+            example: 'fiets'
+        - name: status
```
Comment (remenska) [inline openapi/openapi.yaml:1265]: as a rule of thumb, I move all objects with a schema description of >2loc, or used in more than one place, to the separate schema files.

File: openapi/openapi.yaml [hunk 2]
```diff
@@ -1233,6 +1233,303 @@ paths:
...
+            Oauth-Scope-Required:
+              description: 'The required permission for this particular call'
+              schema:
+                type: string
+                example: 'get_ads'
+          content:
+            application/sellside.ad.count-v1+json:
+              schema:
+                type: integer
+                example: 124
```
Comment (remenska) [inline openapi/openapi.yaml:1342]: Is the response just an integer? I got the impression there was more to it, from this sentence
> Returns a count of ads for the current user and the total size of the result set matching the filter criteria.

File: openapi/openapi.yaml [hunk 3]
```diff
@@ -1233,6 +1233,303 @@ paths:
...
+      security:
+        - bearerAuth: [ ]
+          oAuth2ClientCredentials:
+            - 'console_ro'
+            - 'api_ro'
+  /ad/histogram:
+    get:
+      tags:
+        - Ads
+      summary: 'Get a histogram of ads with filter criteria, grouped by a given field'
```
Comment (remenska) [inline openapi/openapi.yaml:1382]: worth noting that only ads with activity are considered?

File: openapi/openapi.yaml [hunk 4]
```diff
@@ -1233,6 +1233,303 @@ paths:
...
+                example: 'api_ro api_rw'
+            Oauth-Scope-Required:
+              description: 'The required permission for this particular call'
+              schema:
+                type: string
+                example: 'get_ads'
+          content:
+            application/sellside.ad.histogram-v1+json:
+              schema:
+                type: object
```
Comment (remenska) [inline openapi/openapi.yaml:1494]: To make it clear that the values of the map are expected to be int64, worth adding:

```
         additionalProperties:
           type: integer
           format: int64
```

**Review by remenska — DISMISSED** [review]

File: openapi/openapi.yaml (hunk 2 above)
Comment (twessling-icas) [reply openapi/openapi.yaml:1342]: oh, lemme rewrite that then. It is indeed only an integer, as that is OK json.

File: openapi/openapi.yaml (hunk 3 above)
Comment (twessling-icas) [reply openapi/openapi.yaml:1382]: only if timeframe is selected, yes, I'll add that.

File: openapi/openapi.yaml (hunk 4 above)
Comment (twessling-icas) [reply openapi/openapi.yaml:1494]: ah, ok, cool ty! New to this stuff

**Comment by remenska** [comment]
> idk how to properly test this locally except for using the intellij openapi preview pane... any hints to test in real browser from local?

If you already have the openAPI plugin, the linter should indicate any errors/warnings. You can use the openAPI plugin to run a local server, and execute stuff even locally.

Screenshot 2023-11-07 at 10 28 47


If not, you can do a quick&dirty copy/paste in the [Swagger editor](https://editor.swagger.io/) to check for linting errors. I want to add a pre-commit hook eventually.

If you want to _execute_ the API endpoint, you need to authorize to the right env/endpoint (top right button). There are 2 ways to authorize, **both require that you're logged in the console**, i.e., there is a cookie. Either with a test account or just impersonate with readonly.
 1. Use the access token directly with `bearerAuth` - the first option on the screen. Just paste the access token.
 2. login with particular client credentials, requesting the scopes you want. This will create a separate access token, like the normal oauth2 flow that partners use.


I realize now that the caveat of this is that you cannot run your local version against a non-local environment, unless we add localhost to the CORS headers in sellside2. It would first need to be merged to main.

File: openapi/openapi.yaml (hunk 1 above)
Comment (twessling-icas) [reply openapi/openapi.yaml:1265]: oh, I copied this from the GET /ad part..

**Review by remenska — APPROVED** [review]

**Pull request merged**